test/functional/test_framework/test_node.py

```diff
@@ -625,6 +625,9 @@ def add_outbound_p2p_connection(self, p2p_conn, *, wait_for_verack=True, p2p_idx
 
         This method adds the p2p connection to the self.p2ps list and returns
         the connection to the caller.
+
+        p2p_idx must be different for simultaneously connected peers. Reusing it for the next peer
+        after disconnecting the previous one can lead to a race condition and should be avoided.
```

Comment:
nit: the race wouldn't happen with disconnect_p2ps

Reply:
Changed the wording.